### PR TITLE
hide ID fields when creating data sources and reports

### PR DESCRIPTION
**Pull request by nickpell**
https://manage.dimagi.com/default.asp?276990

### CODE REVIEW

**Comment by czue** [comment]
where's the frowning :hankey: emoji when you need it. sorry about this.

File: corehq/apps/userreports/views.py [hunk 1]
```diff
@@ -199,6 +199,7 @@ class UserConfigReportsHomeView(BaseUserConfigReportsView):
 
 class BaseEditConfigReportView(BaseUserConfigReportsView):
     template_name = 'userreports/edit_report_config.html'
+    report_exists = True
```
Comment (czue) [inline corehq/apps/userreports/views.py]: can you not use `self.report_id is not None` for this check?

File: corehq/apps/userreports/views.py [hunk 2]
```diff
@@ -968,6 +970,7 @@ def page_context(self):
 
 class BaseEditDataSourceView(BaseUserConfigReportsView):
     template_name = 'userreports/edit_data_source.html'
+    data_source_exists = True
```
Comment (czue) [inline corehq/apps/userreports/views.py]: (same question as above but for `self.config_id`)

**Review by czue — COMMENTED** [review]
sorry, definitely should have anticipated/caught this in testing. that'll teach me to do quick PID-style features.

provisional approval based on the comments I made not being true / making sense. 

thanks for fixing!

**Review by emord — APPROVED** [review]

**Pull request merged**